bin/node/pallets/pallet-ipfs/src/lib.rs

```diff
@@ -104,11 +104,21 @@ pub mod pallet {
 	#[derive(Clone, Encode, Decode, PartialEq, RuntimeDebug, TypeInfo)]
 	#[scale_info(skip_type_params(T))]
 	pub struct IpfsNode {
-		pub multiaddress: Vec<OpaqueMultiaddr>,
-		pub public_key: Vec<u8>,
-		pub avail_storage: u64,
-		pub files: u64,
-		pub files_total: u64,
+		pub peer_id: Vec<u8>,
+		pub addr: Vec<OpaqueMultiaddr>,
+		pub avail: i32,
+		pub max: i32,
+		pub files: i32,
+	}
```

Comment:
Not sure why you changed naming conventions here. Needs to be discussed.

Reply:
We were trying to work with the same struct for the on-chain storage and the db storage. We had some issues and we left it like that. I suppose we can rename back the `addr` to `multiaddress` and the `peer_id` to `public_key`.

Reply:
> We were trying to work with the same struct for the on-chain storage and the db storage. We had some issues and we left it like that. I suppose we can rename back the `addr` to `multiaddress` and the `peer_id` to `public_key`.

What's wrong with using the same `struct`?

Reply:
Renaming it and using the same seems like a viable solution. So we avoid having two different structs.

Reply:
> 

Okay, we will try that again.

Reply:
> > We were trying to work with the same struct for the on-chain storage and the db storage. We had some issues and we left it like that. I suppose we can rename back the `addr` to `multiaddress` and the `peer_id` to `public_key`.
> 
> What's wrong with using the same `struct`?

We had some issues with the on-chain storage, but we can avoid it now that I took a better look at the code.

Reply:
> Renaming it and using the same seems like a viable solution. So we avoid having two different structs.

Okay! We will work on it.

Reply:
Cool! Waiting for an update.

Reply:
why the type of the `avail`, `max` and `files` are `i32` and not `u32`?

Reply:
> why the type of the `avail`, `max` and `files` are `i32` and not `u32`?

Due to the `diesel` traits on `node-listener`

Reply:
https://kotiri.com/2018/01/31/postgresql-diesel-rust-types.html